ui/app/registration/controllers/editPatientController.js

```diff
@@ -10,10 +10,36 @@ angular.module('bahmni.registration')
             $scope.patient = {};
             $scope.actions = {};
             $scope.myForms = {};
+            $scope.NID = {};
             $scope.addressHierarchyConfigs = appService.getAppDescriptor().getConfigValue("addressHierarchy");
             $scope.disablePhotoCapture = appService.getAppDescriptor().getConfigValue("disablePhotoCapture");
             $scope.today = dateUtil.getDateWithoutTime(dateUtil.now());
             $rootScope.isEligibleForVisit = true;
+
+            var splitNID = function () {
+                return patientService.get(uuid).then(function (response) {
+                    var patientNID = response.patient.identifiers[0].identifier;
+                    var NIDResult = patientNID.split("/");
+                    $scope.codeOfHealthFacility = NIDResult[0];
+                    $scope.NIDYear = parseInt(NIDResult[2], 10);
+                    $scope.NIDsequence = NIDResult[3];
+                });
+            };
+            splitNID();
+
+            $scope.buildFinalNID = function () {
+                if ($scope.NID.healthFacilityCode === undefined) {
+                    $scope.NID.healthFacilityCode = $scope.codeOfHealthFacility;
+                }
+                if ($scope.NID.sequentialCode === undefined) {
+                    $scope.NID.sequentialCode = $scope.NIDsequence;
+                }
+                if ($scope.NID.year === undefined) {
+                    $scope.NID.year = $scope.NIDYear;
+                }
+                $scope.patient.primaryIdentifier.registrationNumber = $scope.NID.healthFacilityCode + '/' + $scope.NID.serviceCode + '/' + $scope.NID.year + '/' + $scope.NID.sequentialCode;
```

Comment:
@simaomazive I believe the error I showed you is coming from here - seems that `$scope.patient.primaryIdentifier` is not always set for a patient